File: 496-Next-Greater-Element-I.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    const solution =[]
    for(let i = 0; i < nums1.length ; i++){
    const findTheindex = (element) => element === nums1[i]
      let elementIndex = nums2.findIndex(findTheindex)
      let nextGreatElement = ""
      for(let j = elementIndex ; j < nums2.length ; j++){
          if(nums2[j] > nums1[i]){
              nextGreatElement = nums2[j]
              break
          }
      }
    nextGreatElement !== "" ? solution.push(nextGreatElement):solution.push(-1)      
    }
    return solution
};

File: 496NextGreaterElementI.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        stack =[]
        
        next_greater = {}
        
        for index, n in enumerate(nums1):
            ind = nums2.index(n)
            temp = nums2[ind:]
            for b in temp:
                if b > n:
                    next_greater[n] = b
                    break
            if not n in next_greater:
                next_greater[n]= -1
            
        for n in nums1:
            stack.append(next_greater[n])
            
        return stack
